code/datums/components/crafting/crafting_external.dm

```diff
@@ -12,21 +12,27 @@
  */
 /atom/proc/CheckParts(list/parts_list, datum/crafting_recipe/R)
 	SEND_SIGNAL(src, COMSIG_ATOM_CHECKPARTS, parts_list, R)
-	if(parts_list)
-		for(var/A in parts_list)
-			if(istype(A, /datum/reagent))
-				if(!reagents)
-					reagents = new()
-				reagents.reagent_list.Add(A)
-				reagents.conditional_update()
-			else if(ismovable(A))
-				var/atom/movable/M = A
-				if(isliving(M.loc))
-					var/mob/living/L = M.loc
-					L.unEquip(M, target = src)
-				else
-					M.forceMove(src)
-				SEND_SIGNAL(M, COMSIG_ATOM_USED_IN_CRAFT, src)
+	if(LAZYLEN(parts_list))
+		for(var/datum/reagent/A as anything in parts_list["reagents"])
```

Comment:
Would it be feasible to pass the reagents datum instead of a list of individual reagents? You could then either `trans_to_holder` or just set myatom on the reagents to the new atom.